parquet-column/src/main/java/org/apache/parquet/column/values/bitpacking/BitPackingValuesWriter.java

```diff
@@ -108,7 +109,11 @@ public void reset() {
 
   @Override
   public void close() {
-    out.close();
+    try (CapacityByteArrayOutputStream cbaos = out) {
+      cbaos.flush();
```

Comment:
This one is used for in-memory encoding only. I suspect similar streams in this module are required to call flush anyway.

Reply:
You want me to remove that ?. For safe side, I am calling flush.. wherever there is buffering possibility

Reply:
I think this PR aims to a potential I/O issue, which now grows larger than I expected. Those output streams used in compression and encoding are mainly as in-memory buffers. So I suspect we should include them as well.

Reply:
@wgtmac Include them .. where. could you pls elaborate more? AFAIK, any outstream which uses in - memory buffers , should be flushed.. When I checked the hierarchy of the Output streams, I found only those which actually needed flush.
For eg: DeltaBinaryPackingValuesWriterForInteger, in close() calls super.close.. And the super.close , has the code for flushing.

Reply:
For example, `CapacityByteArrayOutputStream` class above only inherits the default flush() method, why should we bother calling it here at all?

Reply:
makes sense. it is not needed here.. as of now because as you said CapacityByteArrayOutputStream 's close is sufficient and its flush is a no op.
I can remove it .